app/controllers/address_controller.rb

```diff
@@ -0,0 +1,35 @@
+class AddressController < ApplicationController
+  def create
+    user = User.find_by(verification_token: params[:verification_token])
+    address = user.addresses.build(permitted_address_params)
+
+    if address.save
+      order = logged_in_user_cart_order
+      order.shipping_address_id = address.id
+      order.save
+      redirect_to order_url(logged_in_user_cart_order), notice: t(:successfull, resource_name: 'added shipping address for order')
+    else
+      redirect_to new_address_url, notice: t(:default_error_message)
+    end
+  end
+
+  def new
+    @address = Address.new
+    @order = Order.find(params[:order_id])
+  end
+
+  def shipping
+    order = Order.find(params[:order_id])
```

Comment:
find_by and handle the case when order is not found

Reply:
Created a filter `order_in_params` calling it before :new and :shipping
Querying the using find_by